lib/person.rb

```diff
@@ -21,6 +23,11 @@ def correct_name
     @name
   end
 
+  def rent_book(date, book)
+    new_rental = Rental.new(date, self, book)
+    @rentals << new_rental
+  end
```

Comment:
- [ ] As I mentioned above. The person has many rentals, which means it does contain an array of `rentals` which you did very well but adding the rentals in your function using **<<** is not going to help you because every time anyone tries to add a new person then a new copy of an array will be created. Instead, you can add this `@rental << new_rental` you can add the line in your `Rental.rb` file like `person.rentals << self`. So basically, while adding the new rentals, that rental definitely belongs to a person or a book so that's why you need to add functionality to the `rental.rb` file.

```suggestion
  def rent_book(date, book)
    Rental.new(date, self, book)
  end
```